gpu/matrix/coo_kernels.cpp

```diff
@@ -92,6 +92,15 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_COO_CONJ_TRANSPOSE_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void convert_to_dense(
+    std::shared_ptr<const GpuExecutor> exec, matrix::Dense<ValueType> *result,
+    const matrix::Coo<ValueType, IndexType> *source) NOT_IMPLEMENTED;
```

Comment:
We could for each new format use a default strategy: format->csr->dense. And then eventually replace it with a specific implementation at some point. What do you think @tcojean @gflegar ?

Reply:
This is the general idea behind implementing these `ConvertibleTo<>` interfaces, so that `copy_from` can query which interfaces a certain operator supports, and then choose what to do depending on that - in this case, if we don't have a direct conversion, it can try an indirect one.

If you don't want to get into gory details of exceptions stop reading here.

-----------------

However, it wouldn't work for this specific case where we say we implement `ConvertibleTo<Dense>` interface, and then actually throw a `NotImplemented` error. (You can see how we are actually lying here...).
Of course, we could work around it to also do the right thing even in situations like this, but then we're starting to abuse exceptions to implement control flow (which is perfectly acceptable and even encouraged idiom in Python, but certainly not in C++).
In general, my idea is that running `grep -r "NOT_IMPLEMENTED" *` in any release version of Ginkgo returns only 2 results: NOT_IMPLEMENTED's definition and its unit test.